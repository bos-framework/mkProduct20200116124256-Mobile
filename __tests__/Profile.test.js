import 'react-native';
import React from 'react';
import Profile from '../src/Screens/Container/profileContainer';
import store from '../src/store/store';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<Profile store={store} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
