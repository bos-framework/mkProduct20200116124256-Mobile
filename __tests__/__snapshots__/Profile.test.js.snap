// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#424142",
                "borderColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 100,
                "justifyContent": "center",
                "width": 100,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 40,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 0.7,
            "maxWidth": 300,
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#bdbdbd",
                  "borderRadius": 0,
                  "borderWidth": 1,
                  "marginTop": 15,
                },
                Object {
                  "borderBottomWidth": 1,
                  "borderWidth": 0,
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text>
              First Name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="sentences"
              autoCorrect={false}
              keyboardType="default"
              maxLength={50}
              onChangeText={[Function]}
              placeholder="First Name"
              placeholderTextColor="#919090"
              rejectResponderTermination={true}
              returnKeyType="next"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderRadius": 0,
                    "color": "#545454",
                    "fontSize": 16,
                    "height": 46,
                    "paddingBottom": 0,
                    "paddingLeft": 10,
                    "paddingTop": 2,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "paddingLeft": 0,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#bdbdbd",
                  "borderRadius": 0,
                  "borderWidth": 1,
                  "marginTop": 15,
                },
                Object {
                  "borderBottomWidth": 1,
                  "borderWidth": 0,
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text>
              Last Name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="sentences"
              autoCorrect={false}
              keyboardType="default"
              maxLength={50}
              onChangeText={[Function]}
              placeholder="Last Name"
              placeholderTextColor="#919090"
              rejectResponderTermination={true}
              returnKeyType="next"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderRadius": 0,
                    "color": "#545454",
                    "fontSize": 16,
                    "height": 46,
                    "paddingBottom": 0,
                    "paddingLeft": 10,
                    "paddingTop": 2,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "paddingLeft": 0,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#bdbdbd",
                  "borderRadius": 0,
                  "borderWidth": 1,
                  "marginTop": 15,
                },
                Object {
                  "borderBottomWidth": 1,
                  "borderWidth": 0,
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text>
              Email address
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              editable={false}
              keyboardType="email-address"
              maxLength={50}
              onChangeText={[Function]}
              placeholder="Email Address"
              placeholderTextColor="#919090"
              rejectResponderTermination={true}
              returnKeyType="done"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderRadius": 0,
                    "color": "#545454",
                    "fontSize": 16,
                    "height": 46,
                    "paddingBottom": 0,
                    "paddingLeft": 10,
                    "paddingTop": 2,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "paddingLeft": 0,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#bdbdbd",
                  "borderRadius": 0,
                  "borderWidth": 1,
                  "marginTop": 15,
                },
                Object {
                  "borderBottomWidth": 1,
                  "borderWidth": 0,
                  "marginTop": 10,
                },
              ]
            }
          >
            <Text>
              Phone number
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              keyboardType="phone-pad"
              maxLength={10}
              onChangeText={[Function]}
              placeholder="Phone Number"
              placeholderTextColor="#919090"
              rejectResponderTermination={true}
              returnKeyType="done"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "#dbdbdb",
                    "borderRadius": 0,
                    "color": "#545454",
                    "fontSize": 16,
                    "height": 46,
                    "paddingBottom": 0,
                    "paddingLeft": 10,
                    "paddingTop": 2,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "paddingLeft": 0,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <Text
            style={
              Object {
                "marginTop": 10,
              }
            }
          >
            Data of birth
          </Text>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "width": 142,
                },
                Object {
                  "alignItems": "flex-start",
                  "height": 44,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#aaa",
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "alignItems": "flex-start",
                      "borderBottomWidth": 1,
                      "borderWidth": 0,
                      "height": 44,
                    },
                    false,
                    false,
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  style={
                    Array [
                      Object {
                        "color": "#c9c9c9",
                      },
                      Object {
                        "color": "#545454",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Select a date
                </Text>
              </View>
              <Modal
                animationType="none"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                supportedOrientations={
                  Array [
                    "portrait",
                    "portrait-upside-down",
                    "landscape",
                    "landscape-left",
                    "landscape-right",
                  ]
                }
                transparent={true}
                visible={false}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    clickable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "backgroundColor": "#00000077",
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      clickable={false}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "height": 0,
                            "overflow": "hidden",
                          }
                        }
                      >
                        <View
                          pointerEvents="auto"
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "borderTopColor": "#ccc",
                                  "borderTopWidth": 1,
                                  "marginTop": 42,
                                },
                                undefined,
                              ]
                            }
                          >
                            <RCTDatePicker
                              date={1573209181861}
                              mode="date"
                              onChange={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "height": 216,
                                }
                              }
                              timeZoneOffsetInMinutes={null}
                            />
                          </View>
                        </View>
                        <View
                          accessible={true}
                          clickable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 42,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                                "position": "absolute",
                                "top": 0,
                              },
                              Object {
                                "left": 0,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={true}
                            style={
                              Array [
                                Object {
                                  "color": "#46cf98",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#666",
                                },
                                undefined,
                              ]
                            }
                          >
                            Cancel
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          clickable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 42,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                                "position": "absolute",
                                "top": 0,
                              },
                              Object {
                                "right": 0,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={true}
                            style={
                              Array [
                                Object {
                                  "color": "#46cf98",
                                  "fontSize": 16,
                                },
                                undefined,
                              ]
                            }
                          >
                            Confirm
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </Modal>
            </View>
          </View>
          <Text
            style={
              Object {
                "marginTop": 10,
              }
            }
          >
            Gender
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              clickable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
              testID="ios_touchable_wrapper"
            >
              <TextInput
                allowFontScaling={true}
                editable={false}
                rejectResponderTermination={true}
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "#bdbdbd",
                      "borderWidth": 0,
                      "color": undefined,
                      "fontSize": 16,
                      "height": 44,
                    },
                    Object {
                      "color": "#545454",
                      "fontSize": 16,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value="Select a gender"
              />
            </View>
            <Modal
              hardwareAccelerated={false}
              onDismiss={[Function]}
              onOrientationChange={[Function]}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              testID="ios_modal"
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
                testID="ios_modal_top"
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#EFF1F2",
                      "borderTopColor": "#919498",
                      "borderTopWidth": 0.5,
                      "flexDirection": "row",
                      "height": 44,
                      "justifyContent": "space-between",
                      "paddingHorizontal": 10,
                      "zIndex": 2,
                    },
                    undefined,
                  ]
                }
                testID="input_accessory_view"
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    clickable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "#D0D4DB",
                            "borderRightWidth": 1.5,
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "marginLeft": 11,
                            "transform": Array [
                              Object {
                                "translateY": 4,
                              },
                              Object {
                                "rotate": "-45deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                  <View
                    accessible={true}
                    clickable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "#D0D4DB",
                            "borderRightWidth": 1.5,
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "marginLeft": 22,
                            "transform": Array [
                              Object {
                                "translateY": -5,
                              },
                              Object {
                                "rotate": "135deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  clickable={true}
                  hitSlop={
                    Object {
                      "bottom": 4,
                      "left": 4,
                      "right": 4,
                      "top": 4,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  testID="done_button"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFE",
                          "fontSize": 15,
                          "fontWeight": "bold",
                          "paddingRight": 2,
                          "paddingTop": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    Done
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#D0D4DB",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 215,
                    },
                    undefined,
                  ]
                }
              >
                <View>
                  <RCTPicker
                    items={
                      Array [
                        Object {
                          "label": "Select a gender",
                          "textColor": 4286611584,
                          "value": "",
                        },
                        Object {
                          "label": "Male",
                          "textColor": undefined,
                          "value": "Male",
                        },
                        Object {
                          "label": "Female",
                          "textColor": undefined,
                          "value": "Female",
                        },
                      ]
                    }
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    selectedIndex={0}
                    style={
                      Array [
                        Object {
                          "height": 216,
                        },
                        undefined,
                      ]
                    }
                    testID="ios_picker"
                  />
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "flex": 0.1,
          "marginBottom": 10,
          "maxWidth": 300,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#424142",
            "borderRadius": 2,
            "color": "#fff",
            "height": 46,
            "justifyContent": "center",
            "marginTop": 15,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 17,
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
